### PR TITLE
[WIP] [neon] Optimize sgemm for faster calculation

### PR DESCRIPTION
**Pull request by skykongkong8**
- faster noTrans
- do for every M, K, N

note: there is a slight difference (<1e-2)in transB case, yet

Resolves:

**Self evaluation:**
1. Build test:     [X]Passed [ ]Failed [ ]Skipped
2. Run test:     [X]Passed [ ]Failed [ ]Skipped

### CODE REVIEW

File: nntrainer/tensor/blas_neon.cpp [hunk 1]
```diff
@@ -760,63 +939,109 @@ void sgemm_neon_fp16(const __fp16 *A, const __fp16 *B, __fp16 *C, uint32_t M,
...
+          for (unsigned int idx = n; idx < N; idx++) {
+            C[m * N + idx] = valsC[idx - n];
+          }
         }
       }
     }
+  } else if (!TransA && !TransB) {
+    sgemm_neon_fp16_noTrans(A, B, C, M, N, K, alpha, beta);
   } else { // TransA && TransB
+
```
Comment (SeoHyungjun) [inline nntrainer/tensor/blas_neon.cpp:1044]: It looks like a meaningless line was added.

**Review by SeoHyungjun — APPROVED** [review]
LGTM!

**Review by baek2sm — APPROVED** [review]
LGTM

**Comment by skykongkong8** [comment]
Close to make a updated pr

**Pull request closed without merging**